infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -96,6 +96,14 @@ public class CorfuReplicationDiscoveryService implements Runnable, CorfuReplicat
      * System exit error code called by the Corfu Runtime systemDownHandler
      */
     private static final int SYSTEM_EXIT_ERROR_CODE = -3;
+
+    /**
+     * Used for managing the set of streams to replicate, and also used for upgrading path
+     * in Log Replication
+     */
+    @Getter
+    private static LogReplicationStreamNameTableManager replicationStreamNameTableManager;
```

Comment:
It doesn't make much sense for this to be a static.. would it make more sense to pass it down to whichever layer needs it? Probably through the: CorfuReplicationManager...